solidity/contracts/BancorNetwork.sol

```diff
@@ -627,7 +629,56 @@ contract BancorNetwork is IBancorNetwork, TokenHolder, ContractIds, FeatureIds {
     }
 
     /**
-        @dev utility, checks whether allowance for the given spender exists and approves one if it doesn't
+        @dev ensures transfer of tokens, taking into account that some ERC-20 implementations don't return
+        true on success but revert instead
+
+        @param _token     the token to transfer
+        @param _to        the address to transfer the tokens to
+        @param _amount    the amount to transfer
+    */
+    function ensureTransfer(IERC20Token _token, address _to, uint256 _amount) private {
+        ITokenWhitelist tokenWhitelist = ITokenWhitelist(registry.addressOf(ContractIds.TOKEN_WHITELIST));
+
+        if (!tokenWhitelist.whitelistedTokens(_token)) {
+            // if the token isn't whitelisted, we assert on transfer
+            assert(_token.transfer(_to, _amount));
+        } else {
+            uint256 prevBalance = _token.balanceOf(_to);
+            // we have to cast the token contract in an interface which has no return value
+            INonStandardERC20(_token).transfer(_to, _amount);
+            uint256 postBalance = _token.balanceOf(_to);
+            assert(postBalance.sub(prevBalance) == _amount);
```

Comment:
that might not work for tokens that have fees, maybe just check that the new amount is greater than the prev amount?

Reply:
hmmm, didn't consider that case... we should discuss this further